cli/azd/pkg/pipeline/azdo_provider.go

```diff
@@ -724,7 +766,12 @@ func (p *AzdoCiProvider) configureConnection(
 	authType PipelineAuthType,
 	credentials *azcli.AzureCredentials,
 ) error {
+	if authType == "" || authType == AuthTypeFederated {
```

Comment:
Just wondering - in what cases do we call this with an authType of `""`?  My naive belief is that we should only ever call this with either `AuthTypeFederated` or `AuthTypeClientCredentials`, and that if it's called with "" that's a similar error to calling it with something like `PipelineAuthType("bogus")`

If there's actually a case where we wanted to reflect "A choice has not been made" and we were using `""` for that, I'd almost rather us change `authType PipelineAuthType` to `authType *PipelineAuthType` and allow `nil` to have the special meaning that `""` has today.

Reply:
authType == "" is the default value for the command's flag:

![image](https://github.com/Azure/azure-dev/assets/24213737/adbcc6ad-44e4-4a51-8f70-d1fbea6cd6b0)

We can totally change the default to federatedCredential.  I think we did not set a default for the flag because the default depends on the configuration as:

terraform + [gh or azdo] => client secret
bicep + azdo => client secret
bicep + gh => federated cred

After this PR, the default value changes to:

terraform + [gh or azdo] => client secret
bicep + azdo => **federated cred**
bicep + gh => federated cred

Similiar to my previous comment, I can use a future PR for terraform to correct the default and remove those empty string conditions.